framework/tests/src/tests/signatureinference/SignatureInferenceTestAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,176 @@
+import tests.signatureinference.qual.Sibling1;
+import tests.signatureinference.qual.Sibling2;
+import tests.signatureinference.qual.SiblingWithFields;
+import tests.signatureinference.qual.SignatureInferenceBottom;
+import tests.signatureinference.qual.Top;
+/**
+ * AnnotatedTypeFactory to test signature inference using .jaif
+ * files.
+ * <p>
+ * The used qualifier hierarchy is straightforward and only intended for test
+ * purposes.
+ * 
+ * @author pbsf
+ */
+public class SignatureInferenceTestAnnotatedTypeFactory
+        extends
+            BaseAnnotatedTypeFactory {
+
+    private final AnnotationMirror PARENT = new AnnotationBuilder(
+            processingEnv, Parent.class).build();
+    private final AnnotationMirror BOTTOM = new AnnotationBuilder(
+            processingEnv, SignatureInferenceBottom.class).build();
+
+    public SignatureInferenceTestAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(checker);
+        postInit();
+        addTypeNameImplicit(java.lang.Void.class, BOTTOM);
+    }
+
+    @Override
+    protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+        return Collections.unmodifiableSet(
+                new HashSet<Class<? extends Annotation>>(Arrays.asList(
+                    Parent.class, DefaultType.class, Top.class, Sibling1.class,
+                    Sibling2.class, SignatureInferenceBottom.class,
+                    SiblingWithFields.class, ImplicitAnno.class)));
+    }
+    @Override
+    public TreeAnnotator createTreeAnnotator() {
+        ImplicitsTreeAnnotator implicitsTreeAnnotator = new ImplicitsTreeAnnotator(
+                this);
+        implicitsTreeAnnotator.addTreeKind(
+                com.sun.source.tree.Tree.Kind.NULL_LITERAL, BOTTOM);
+        implicitsTreeAnnotator.addTreeKind(
+                com.sun.source.tree.Tree.Kind.INT_LITERAL, BOTTOM);
+
+        return new ListTreeAnnotator(new PropagationTreeAnnotator(this),
+                implicitsTreeAnnotator);
+    }
+
+    @Override
+    public QualifierHierarchy createQualifierHierarchy(MultiGraphFactory factory) {
+        return new SignatureTestQualifierHierarchy(factory);
+    }
+
+    /**
+     * Using a MultiGraphQualifierHierarchy to enable tests with Annotations
+     * that contain fields. @see SiblingWithFields.
+     * @author pbsf
+     *
+     */
+    protected class SignatureTestQualifierHierarchy extends
+                MultiGraphQualifierHierarchy {
+
+        public SignatureTestQualifierHierarchy(MultiGraphFactory f) {
+            super(f);
+        }
+
+        @Override
+        public AnnotationMirror getBottomAnnotation(AnnotationMirror start) {
+            return BOTTOM;
+        }
+
+        @Override
+        public AnnotationMirror leastUpperBound(AnnotationMirror a1,
+                AnnotationMirror a2) {
+            if ((AnnotationUtils.areSameByClass(a1, Sibling1.class)
+                 && AnnotationUtils.areSameByClass(a2, Sibling2.class)) 
```

Comment:
Why is the indentation only different by a single character? What do the operators align with?